allennlp/training/trainer.py

```diff
@@ -242,6 +266,10 @@ class GradientDescentTrainer(Trainer):
         precision training. Must be a choice of `"O0"`, `"O1"`, `"O2"`, or `"O3"`.
         See the Apex [documentation](https://nvidia.github.io/apex/amp.html#opt-levels-and-properties) for
         more details. If `None`, Amp is not used. Defaults to `None`.
+    batches_per_epoch: `int`, optional, (default = `None`)
```

Comment:
We are trying to minimize the parameters that need to passed directly to the trainer.  I'm pretty sure you don't even need this, as you can just specify a `DataLoader`.  Can you make your data loader class inherit from our `DataLoader`, and register it as something?  In this PR, I'd like to see only modifications to data code, with no changes necessary to the trainer at all.

Reply:
How do you like if I just add this optional parameter to existing Allennlp ```DataLoader```, so that default custom ```collate_fn``` is not the only reason we subclass it?

Reply:
Sure, that seems reasonable enough to me.

Reply:
In this case, instead of specifying the new data loader with "type" keyword, just in order to be able to add a single parameter to the config file, the users will just get it as an additional field for free.